haskell/test/Test/Ligo/LambdaDAO.hs

```diff
@@ -0,0 +1,255 @@
+
+import Lorentz as L hiding (assert, div)
+import Lorentz.Contracts.Spec.FA2Interface qualified as FA2
+import Morley.Michelson.Typed (convertContract)
+import Morley.Michelson.Typed.Convert (untypeValue)
+import Morley.Util.Named
+import Test.Cleveland
+
+import Ligo.BaseDAO.Contract
+import Ligo.BaseDAO.ErrorCodes
+import Ligo.BaseDAO.LambdaDAO.Types
+import Ligo.BaseDAO.Types
+import Test.Ligo.BaseDAO.Common
+
+getStorageRPCLambda :: forall p caps vd m. MonadCleveland caps m => TAddress p vd ->  m (StorageSkeletonRPC (VariantToExtra 'Lambda))
+getStorageRPCLambda addr = getStorage @(StorageSkeleton (VariantToExtra 'Lambda)) (unTAddress addr)
+
+withOriginated
+  :: MonadCleveland caps m
+  => Integer
+  -> ([Address] -> LambdaStorage)
+  -> ([Address] -> LambdaStorage -> TAddress (Parameter' LambdaCustomEpParam) () -> TAddress FA2.Parameter () -> m a)
+  -> m a
+withOriginated addrCount storageFn tests = do
+  addresses <- mapM (\x -> newAddress $ fromString ("address" <> (show x))) [1 ..addrCount]
+  dodTokenContract <- chAddress <$> originateSimple "token_contract" [] dummyFA2Contract
+  let storageInitial = storageFn addresses
+  now_level <- getLevel
+  let storage = storageInitial
+        { sGovernanceToken = GovernanceToken
+              { gtAddress = dodTokenContract
+              , gtTokenId = FA2.theTokenId
+              }
+          , sStartLevel = now_level
+        }
+
+  baseDao <- originateUntyped $ UntypedOriginateData
+    { uodName = "BaseDAO - LambdaDAO Test Contract"
+    , uodBalance = zeroMutez
+    , uodStorage = untypeValue $ toVal storage
+    , uodContract = convertContract baseDAOLambdaLigo
+    }
+  tests addresses storage (TAddress baseDao) (TAddress dodTokenContract)
+
+toPeriod :: LambdaStorage -> Natural
+toPeriod = unPeriod . cPeriod . sConfig
+
+-- | A proposal handler check that check the
+-- input string is not empty.
+proposal_1_check :: HandlerCheck
+proposal_1_check =
+  L.car #
+  (unpackRaw @MText) #
+  assertSome [mt|Input cannot be unpacked|] #
+  L.size #
+  assertNeq0 [mt|Input cannot be empty|] #
+  push ()
+
+
+-- | A proposal handler that accept a text value
+-- and inserts it into a list in handler storage.
+proposal_1_handler :: ProposalHandler
+proposal_1_handler =
+  (getField #piPacked_argument) #
+  (unpackRaw @MText) #  ifNone dummyHandler insertHandlerStorage
+  where
+    insertHandlerStorage :: '[MText, PhInput] :-> '[PhOutput]
+    insertHandlerStorage =
+      dip (toField #piHandler_storage) #
+      packRaw #
+      push [mt|key0|] #
+      mapInsert #
+      L.nil @Operation #
+      L.swap #
+      L.none #
+      constructStack @PhOutput
+
+dummyHandler :: '[PhInput] :-> '[PhOutput]
+dummyHandler =
+  toField #piHandler_storage #
+  L.nil @Operation #
+  L.swap #
+  L.none #
+  constructStack @PhOutput
+
+-- | We test non-token entrypoints of the BaseDAO contract here
+test_LambdaDAO :: [TestTree]
+test_LambdaDAO =
+  [ testGroup "LambdaDAO Tests"
```

Comment:
I think we should also add test cases for when:
- `remove_handler` call succeed
- After `remove_handler` call, the proposal that has not been flush should be still allow to execute the handler
- After `remove_handler` call, a new `execute_handler` proposal shouldn't be allowed to use that handler.